generators/server/templates/src/main/java/package/security/oauth2/CustomClaimConverter.java.ejs

```diff
@@ -83,6 +83,30 @@ public class CustomClaimConverter implements Converter<Map<String, Object>, Map<
                 if (user.has("family_name")) {
                     convertedClaims.put("family_name", user.get("family_name").asText());
                 }
+                if (user.has("email")) {
+                    convertedClaims.put("email", user.get("email").asText());
+                }
+                // Allow full name in a name claim - happens with Auth0
+                if (user.has("name")) {
+                    String[] name = user.get("name").asText().split("\\s+");
+                    if (name.length > 0) {
+                        convertedClaims.put("given_name", name[0]);
+                        if (name.length == 2) {
+                            convertedClaims.put("family_name", name[1]);
+                        } else {
+                            StringBuilder lastName = new StringBuilder();
+                            int i = 1;
+                            while (i < name.length) {
+                                lastName.append(name[i]);
+                                i++;
+                                if (i < name.length) {
+                                    lastName.append(" ");
+                                }
+                            }
+                            convertedClaims.put("family_name", lastName.toString());
+                        }
```

Comment:
for `length > 1`, you can probably write as `String.join(", ", Arrays.asList(name).subList(1, name.length).toArray(new String[0]))`

Reply:
I'm a bit hesitant to change things now. However, I'd be happy to accept your changes if you modify this PR.

Reply:
I have updated comment with suggested replacement. If you agree, you can merge the change.

Reply:
IntelliJ suggested an even shorter version with `Arrays.copyOfRange`. I just committed it.

Reply:
ok. I just wanted to avoid creation of a new array. Considering size of input array, I think copy should also be fine